### PR TITLE
Fee pooling SDK change

### PR DESCRIPTION
**Pull request by shiqizng**
This PR is to support creating transactions with a fee under mintxnfee since other txns in the group will be paying.

### CODE REVIEW

File: future/transaction_test.go [hunk 1]
```diff
@@ -741,10 +742,10 @@ func TestFee(t *testing.T) {
 			expected: 1001,
 		},
 		{
-			name:     "Flat fee overridden with min fee",
+			name:     "Flat fee does not overridden with min fee",
```
Comment (jdtzmn) [inline future/transaction_test.go:745]: nit: _is not_ or _does not get_

**Review by jdtzmn — CHANGES_REQUESTED** [review]
Mostly looks good - I just left one comment. Can you also add a test case that explicitly sets the fee to 0?

**Review by jasonpaulos — CHANGES_REQUESTED** [review]
Implementation looks good, but could you modify the descriptions for the `SuggestedParams` struct? Specifically:
* For `Fee`, it current says "Fee may fall to zero but transactions must still have a fee of at least MinTxnFee for the current network protocol." Maybe change this to "Fee may fall to zero but a group of N atomic transactions must still have a fee of at least N*MinTxnFee for the current network protocol."
* For `FlatFee`, mention that if this is true, the fee is not guaranteed to be above the network minimum.

**Review by jasonpaulos — APPROVED** [review]
Looks good! Thanks for making the requested changes

File: types/transaction.go [hunk 2]
```diff
@@ -162,8 +162,8 @@ type TxGroup struct {
 type SuggestedParams struct {
 	// Fee is the suggested transaction fee
 	// Fee is in units of micro-Algos per byte.
-	// Fee may fall to zero but transactions must still have a fee of
-	// at least MinTxnFee for the current network protocol.
+	// Fee may fall to zero but a group of N atomic transactions must
+	// still have a fee of at least N*MinTxnFee for the current network protocol.
```
Comment (winder) [inline types/transaction.go:166]: Might be nice to check the minimum balance in `crypto.ComputeGroupID`

File: types/transaction.go [hunk 3]
```diff
@@ -183,6 +183,7 @@ type SuggestedParams struct {
 	ConsensusVersion string
 
 	// FlatFee indicates whether the passed fee is per-byte or per-transaction
+	// If true, txn fee is may fall below the MinTxnFee for the current network protocol.
```
Comment (jdtzmn) [inline types/transaction.go:186]: nit: txn fee _may_ fall

**Review by jdtzmn — CHANGES_REQUESTED** [review]

**Review by jdtzmn — APPROVED** [review]

**Pull request merged**